### PR TITLE
Replace external links with internal ones where appropriate and fix internal links

### LINKED ISSUE #2551: Cannot load infomation in "Multi-Factor Authenticators" popup.
### Bug description

When I access "Knowlegde Base" => "Introduction to Passwords" section. It does not show infomation of "multi-factor authentication" popup. At least, it should show 2 first line titiles they are

"Hardware Security Keys
YubiKey
...."

You can refer attached photo for more details.

IMG_6136

### Affected pages

-

### Browser

_No response_

### Operating System

_No response_

### Before submitting

- [X] I am reporting something that is broken on the website, not making a suggestion.
- [X] I agree to the [Community Code of Conduct](https://www.privacyguides.org/coc).

### PR DESCRIPTION
**Pull request by redoomed1**
Changes proposed in this PR:

- Replace several external links for tools already listed on the site with internal links if an embedded link points to a website already listed on the relevant recommendation card
- On the DNS Overview page, expand on introduction for "What is Encrypted DNS?" for more useful instant previews
  - Internal links found [here](https://www.privacyguides.org/en/basics/common-threats/#avoiding-censorship) and [here](https://www.privacyguides.org/en/advanced/dns-overview/#what-is-dns)
- On the Multi-Factor Authentication page, fix internal link to VeraCrypt card
- `tools.md`
  - Add Redlib and ProxiTok entries as it seems that the PRs that added both did not do so
  - Fix link to OS Full Disk Encryption section under the teal "Operating System Disk Encryption" toggle

<summary>

- [x] I agree to the terms listed below:
  <details><summary>Contribution terms (click to expand)</summary>
  1) I am the sole author of this work.
  2) I agree to grant Privacy Guides a perpetual, worldwide, non-exclusive, transferable, royalty-free, irrevocable license with the right to sublicense such rights through multiple tiers of sublicensees, to reproduce, modify, display, perform, relicense, and distribute my contribution as part of this project.
  3) I have disclosed any relevant conflicts of interest in my post.
  4) I agree to the Community Code of Conduct.
  </details>

### CODE REVIEW

**Comment by redoomed1** [comment]
I would greatly appreciate any feedback on the new intros added to the Multi-Factor Authenticator pages.

My approach while drafting these was to add the minimal amount of information needed to understand the tools on the page and their importance, then add links that direct readers to the relevant sections in the Knowledge Base for more information.

File: docs/advanced/dns-overview.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ An observer could modify any of these packets.
 
 ## What is "encrypted DNS"?
 
-Encrypted DNS can refer to one of a number of protocols, the most common ones being:
+Encrypted DNS can refer to one of a number of protocols, the most common ones being DNSCrypt, DNS over TLS (DoT), and DNS over HTTPS (DoH). We discuss each protocol's implementation and level of operating system support below.
```
Comment (jonaharagon) [inline docs/advanced/dns-overview.md:69]: what if we did this instead

```suggestion
Encrypted DNS can refer to one of a number of protocols, the most common ones being [DNSCrypt](#dnscrypt), [DNS over TLS](#dns-over-tls-dot), and [DNS over HTTPS](#dns-over-https-doh).
```

**Comment by dngray** [comment]
This one looks good. Are you ready for merging?

**Comment by redoomed1** [comment]
Yes @dngray

**Pull request merged**